homeassistant/components/fan/__init__.py

```diff
@@ -235,6 +235,13 @@ class FanEntity(ToggleEntity):
     """Base class for fan entities."""
 
     entity_description: FanEntityDescription
+    _attr_current_direction: str | None = None
+    _attr_oscillating: bool | None = None
+    _attr_percentage: int | None = None
+    _attr_preset_mode: str | None = None
+    _attr_preset_modes: list[str] | None = None
+    _attr_speed_count: int | None = None
+    _attr_supported_features = 0
```

Comment:
The default value isn't the only change from upstream, we need to re-type it as well, as `None` is not a valid supported features value for the fan platform.

```suggestion
    _attr_supported_features: int = 0
```